src/lib/layout-context.ts

```diff
@@ -170,3 +180,24 @@ const normalizePropertyFormatForDocs = (format: PropertyFormat): string => {
 
   return formatMap[format] ?? pascalCase(format)
 }
+
+const flattenObjectProperties = (
+  properties: Property[],
+  namespace: string = '',
+): Property[] => {
+  const results: Property[] = []
+
+  for (const property of properties) {
+    const name =
+      namespace === '' ? property.name : `${namespace}.${property.name}`
+
+    results.push({ ...property, name })
+
+    if (property.format === 'object') {
+      results.push(...flattenObjectProperties(property.properties, name))
+      continue
+    }
+  }
+
+  return results
+}
```

Comment:
Namespace is already "reserved". Avoid assigning special meaning to the empty string: prefer an explicit null, or this empty array pattern.
```suggestion
const flattenObjectProperties = (
  properties: Property[],
  paths: string[] = [],
): Property[] => {
  const results: Property[] = []

  for (const property of properties) {
    const name = [...path, property.name].join('.')

    results.push({ ...property, name })

    if (property.format === 'object') {
      results.push(...flattenObjectProperties(property.properties, [name]))
      continue
    }
  }

  return results
}
```